### DISCUSSION TITLE
Autoplay on iOS

### DISCUSSION THREAD

**Discussion by vettorazi** [gilbarbara/react-spotify-web-playback#137] [Q&A] [answered]
The player is working great on the desktop, but it seems that on ios, the autoplay or play props don't work.
I see that sometimes the icon of pause changes to the play for a brief second and then go back to the previous icon.

This is something related to IOS or it's possible to have this player playing automatically after the component loads?

**Comment by vettorazi** [comment]
More info about this issue:
the button of play isn't touchable when the component loads. I have to skip the next song to the music starts.
after that, everything works fine.
my component seems pretty simple:
```
            <SpotifyPlayer
              name="My Player"
              token={playerToken}
              uris={songs[index].songID}
              autoPlay={true}
              styles={{
                bgColor: "transparent",
              }}
            />

```

**Comment by gilbarbara** [comment, marked as answer]
It's a long-standing limitation on iOS.
https://developer.spotify.com/documentation/web-playback-sdk/#mobile

**Comment by vettorazi** [comment]
Yeah, I tried to add a button that controls the state of the play prop of the component and that also didn’t work. 
what’s the difference from pressing my button, that controls the play state (which doesn’t work before I press yours) than pressing the play button of your player?

**Reply by gilbarbara** [reply]
Hey @vettorazi 

Can you try again with v0.13.0?
The `autoPlay` is deprecated now since it also stopped working on the Chrome desktop. So I won't support it anymore.
You need user interaction on the page for the "play" prop to work.

**Reply by gilbarbara** [reply]
> what’s the difference from pressing my button, that controls the play state (which doesn’t work before I press yours) than pressing the play button of your player?

The player play/pause button controls the SDK directly while the change in the `play` prop tries to emulate that.
